### PR TITLE
feat: support emotion 10 & remove hacks

### PR DESCRIPTION
**Pull request by neoziro**
## BREAKING CHANGES:

- `OriginalComponent.withComponent(NewComponent)` is replaced by `uiAs(OriginalComponent, NewComponent)`

- `as={NewComponent}` is replaced by `uiAs={NewComponent}`

- `globalStyle()` is now replaced by `Normalize` component

### CODE REVIEW

File: packages/system/src/util.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    .split('.')
+    .reduce((a, b) => (a && is(a[b]) ? a[b] : undefined), obj)
+
+export function merge(acc, item) {
+  if (!item) {
+    return acc
+  }
+
+  return deepmerge(acc, item, {
+    clone: false, // No need to clone deep, it's way faster.
```
Comment (oliviertassinari) [inline packages/system/src/util.js]: ;)

File: packages/system/src/style.js [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+    }
+  })
+
+  const generatorsByProp = indexGeneratorsByProp(flatGenerators)
+
+  function getStyle(attrs, theme) {
+    const propKeys = Object.keys(attrs)
+    const propCount = propKeys.length
+    let allStyle = {}
+    for (let i = 0; i < propCount; i++) {
```
Comment (oliviertassinari) [inline packages/system/src/style.js]: for > map :)

File: packages/system/src/util.js (hunk 1 above)
Comment (neoziro) [reply packages/system/src/util.js]: Inspired by you!

File: packages/system/src/style.js (hunk 2 above)
Comment (neoziro) [reply packages/system/src/style.js]: Complexity n, even with `compose` ;)

**Pull request merged**